47.md

```diff
@@ -63,6 +63,267 @@ Response:
 }
 ```
 
+Below are the commands that are defined in this NIP.
+
+### Pay Invoice
+
+Request:
+```jsonc
+{
+    "method": "pay_invoice",
+    "params": {
+        "invoice": "lnbc50n1..." // invoice to pay, required
```

Comment:
many use cases (zaps, boosts, etc.) use split payments and I have heard this requirement quite often. 
could we optionally allow an array in in the `pay_invoice` and `pay_keysend` methods? 
This would reduce the amount of calls that have to be made in those cases. 

As result we would then also return an array

Reply:
Yeah have heard this requested a lot

Reply:
One problem is if you aren't able to pay all of them, how do you handle whether or not to send an error or success message

Reply:
I think this would be better as a separate command. Maybe leave for future work as this PR has lots of changes already

Reply:
ok, you mean doing it in a `multi_pay_invoice` command or something like that? 
would be OK for me.

Reply:
> One problem is if you aren't able to pay all of them, how do you handle whether or not to send an error or success message

There could be one payment request event for 20 invoices, but 20 response events for each individual payment (each tagged with the request event ID, just different preimages/errors in the content). 
Allowing for an array of invoices would mean zap split payments don't have to spam the nwc relays.

Reply:
@gudnuf this is perfect! 
we probably then should just add the payment_hash or the invoice itself to the result? 

@benthecarman what do you think? is that good? 

split payments has been used quite a lot so I think it would be nice to have a decent support here.

Reply:
The challenge with that is attributing which response corresponds to which invoice. We could tag the event with the payment hash i guess